kratos/python/add_constraint_to_python.cpp

```diff
@@ -114,6 +114,10 @@ void AddConstraintToPython(pybind11::module &m)
     .def("SetValue", SetValueHelperFunctionMasterSlaveConstraint< Variable< std::string > >)
     .def("GetValue", GetValueHelperFunctionMasterSlaveConstraint< Variable< std::string > >)
 
+    .def("GetMasterDofsVector", &MasterSlaveConstraint::GetMasterDofsVector, pybind11::return_value_policy::reference_internal)
```

Comment:
Why is the special pybind return value policy necessary?

Reply:
Otherwise the dofs get deleted when the python vector gets out of scope (at least it segfaults...)

Reply:
As far as i see pybind creates a python list with the pointers inside.
```
[<Kratos.Dof object at 0x7ffa55a9cb30>, <Kratos.Dof object at 0x7ffa55a9cb70>, ...]
```

I think it is the same reason why e.g. it is needed for the NodesArray() in the model part has this policy:
```cpp
        .def("NodesArray", &ModelPart::NodesArray, py::return_value_policy::reference_internal)
        .def_property("Nodes", ModelPartGetNodes1, ModelPartSetNodes1)
        .def("GetNodes", ModelPartGetNodes1)
```
where `NodesArray` returns the actual container, while `Nodes` gives the PointerVectorSet. but to be honest i am lost in the typdefs.

Reply:
EDIT: according to pybind docs this is also the default for `def_property` getters.